ThirdPartyModules/Mirasvit/Credit.php

```diff
@@ -0,0 +1,190 @@
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace Bolt\Boltpay\ThirdPartyModules\Mirasvit;
+
+use Bolt\Boltpay\Helper\Discount;
+use Bolt\Boltpay\Helper\Shared\CurrencyUtils;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Sales\Model\Service\OrderService;
+use Magento\Framework\App\State;
+
+class Credit
+{
+    /**
+     * @var State
+     */
+    private $appState;
+    
+    /**
+     * @var DiscountHelper
+     */
+    protected $discountHelper;
+    
+    /**
+     * @var \Mirasvit\Credit\Helper\Data
+     */
+    protected $mirasvitStoreCreditHelper;
+    
+    /**
+     * @var \Mirasvit\Credit\Helper\Calculation
+     */
+    protected $mirasvitStoreCreditCalculationService;
+    
+    /**
+     * @var \Mirasvit\Credit\Api\Config\CalculationConfigInterface
+     */
+    protected $mirasvitStoreCreditCalculationConfig;
+    
+    /**
+     * @var \Mirasvit\Credit\Service\Config\CalculationConfig
+     */
+    protected $mirasvitStoreCreditCalculationConfigLegacy;
+
+    /**
+     * @param Discount                     $discountHelper
+     * @param EventsForThirdPartyModules   $eventsForThirdPartyModules
+     * @param State                        $appState
+     */
+    public function __construct(
+        Discount    $discountHelper,
+        State       $appState
+    ) {
+        $this->discountHelper = $discountHelper;
+        $this->appState       = $appState;
+    }
+
+    public function collectDiscounts($result,
+                                     $mirasvitStoreCreditHelper,
+                                     $mirasvitStoreCreditCalculationService,
+                                     $mirasvitStoreCreditCalculationConfig,
+                                     $mirasvitStoreCreditCalculationConfigLegacy,
+                                     $quote,
+                                     $paymentOnly)
+    {
+        list ($discounts, $totalAmount, $diff) = $result;
+        $this->mirasvitStoreCreditHelper = $mirasvitStoreCreditHelper;
+        $this->mirasvitStoreCreditCalculationService = $mirasvitStoreCreditCalculationService;
+        $this->mirasvitStoreCreditCalculationConfig = $mirasvitStoreCreditCalculationConfig;
+        $this->mirasvitStoreCreditCalculationConfigLegacy = $mirasvitStoreCreditCalculationConfigLegacy;
+
+        try {
+            $amount = abs($this->getMirasvitStoreCreditAmount($quote, $paymentOnly));
+            $currencyCode = $quote->getQuoteCurrencyCode();
+            $roundedAmount = CurrencyUtils::toMinor($amount, $currencyCode);
+
+            $discounts[] = [
+                'description'       => 'Store Credit',
+                'amount'            => $roundedAmount,
+                'discount_category' => Discount::BOLT_DISCOUNT_CATEGORY_STORE_CREDIT,
+                'discount_type'     => $this->discountHelper->getBoltDiscountType('by_fixed'), // For v1/discounts.code.apply and v2/cart.update
```

Comment:
This class it too big to include in only for using this helper. How about moving into static method, the same as `CurrencyUtils::toMinorWithoutRounding`?
It's ok for me if we just change `getBoltDiscountType` to static and leave it in the same class.

Reply:
It make sense, but to change `getBoltDiscountType` to static would affect a lot of other code which seems to be irrelevant to this task, then shall we do it in another PR?

Reply:
Yes, another PR makes sense. Thank you Ning.